system/security/cryptohelper/ske.cpp

```diff
@@ -0,0 +1,161 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+/*
+ * Module dealing with openssl symmetric key encryption/decription.
+ * For now just AES.
+ */
+
+#if defined(_USE_OPENSSL) && !defined(_WIN32)
+
+#include "jliball.hpp"
+#include <openssl/pem.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+
+#include "cryptocommon.hpp"
+#include "pke.hpp"
+#include "ske.hpp"
+
+namespace cryptohelper
+{
+
+size32_t aesKeyEncrypt(MemoryBuffer &out, size32_t inSz, const void *inBytes, const char key[aesKeySize], const char iv[aesBlockSize])
```

Comment:
Slightly unusual to prototype a function argument as a fixed size array (I spent a while googling to see if it has any extra semantics in c++11).  May be useful for documentation,  the size will not be checked.

Reply:
I think I used it, because the openssl EVP examples were..